03-rnaseq/differential-expression_rnaseq_01.Rmd

```diff
@@ -0,0 +1,419 @@
+
+## Place the dataset in your new `data/` folder
+
+Refine.bio will send you a download button in the email when it is ready. 
+Follow the prompt to download a zip file that has a name with a series of letters and numbers and ends in `.zip`. 
+Double clicking should unzip this for you and create a folder of the same name.  
+
+<img src="https://github.com/AlexsLemonade/refinebio-examples/raw/8f1332ce8c5deb0dd93edf6c84b93bae676d9f3b/template/screenshots/download-folder-structure.png" width=400> 
+
+For more details on the contents of this folder see [these docs on refine.bio](http://docs.refine.bio/en/latest/main_text.html#rna-seq-sample-compendium-download-folder).  
+
+The `<experiment_accession_id>` folder has the data and metadata TSV files you will need for this example analysis.
+Experiment accession ids usually look something like `GSE1235` or `SRP12345`. 
+
+Copy and paste the `<experiment_accession_id>` folder into your newly created `data/` folder.
+
+## Check out our file structure!
+
+Your new analysis folder should contain: 
+
+- The example analysis `.Rmd` you downloaded  
+- A folder called "data" which contains:
+  - The `<experiment_accession_id>` folder which contains:
+    - The gene expression  
+    - The metadata TSV  
+
+Your example analysis folder should now look something like this (except with respective experiment accession id and analysis notebook name you are using): 
+
+<img src="https://github.com/AlexsLemonade/refinebio-examples/raw/b78046845763272d0aa65a077b6a3e46977d5342/template/screenshots/analysis-folder-structure.png" width=400>
+
+In order for our example here to run without a hitch, we need these files to be in these locations so we've constructed a test to check before we get started with the analysis. 
+Run this chunk to double check that your files are in the right place. 
+
+```{r}
+# Check if the gene expression matrix file is in the data directory stored as `data_dir`
+file.exists(file.path(data_dir, "SRP078441.tsv"))
+
+# Check if the metadata file is in the data directory stored as `data_dir`
+file.exists(file.path(data_dir, "metadata_SRP078441.tsv"))
+```
+
+If the chunk above printed out `FALSE` to either of those tests, you won't be able to run this analysis _as is_ until those files are in the appropriate place.
+
+If the concept of a "file path" is unfamiliar to you; we recommend taking a look at our [section about file paths](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#an-important-note-about-file-paths-and-Rmds). 
+
+# Using a different refine.bio dataset with this analysis?
+
+If you'd like to adapt an example analysis to use a different dataset from [refine.bio](https://www.refine.bio/), we recommend placing the files in the `data/` directory you created and changing the filenames and paths in the notebook to match these files (we've put comments to signify where you would need to change the code).
```

Comment:
Hmm... I do like this idea, but also am worried about people being confused about too much extra comments and hashtags when they are first going through the notebook. I feel like it may be visually distracting or conceptually confusing for particularly new to R users. 

But if you want to show an example of what this could look like, I'm open to seeing if there's a way we can make the "changeme" comments pop. (Could file a separate issue for discussion if you like).

Reply:
I like where this is going... could it be achieved by not hardcoding SRPwhatever into those file names but using a variable name to construct the output file paths? The variable name could get defined once at the top of the script and this would be the only place to highlight it/change it?

Reply:
There's more than just file names that folks might need to change when personalizing their analyses, and I'm concerned about having too many variables for them to keep track of. But this is also an interesting idea that we might be able to use in some example analyses.